app/models/user.rb

```diff
@@ -384,21 +392,14 @@ def announcement_receiver(target)
       end
     end
 
-    def users_role(target)
-      case target
-      when 'student_and_trainee'
-        students_and_trainees
-      when 'graduate'
-        graduated
-      when 'adviser'
-        advisers
-      when 'inactive'
-        inactive.order(:last_activity_at)
-      when 'trainee'
-        trainees
-      else
-        send(target)
-      end
+    # このメソッドはユーザから送信された値をsendに渡すので、悪意のあるコードが実行される危険性がある
+    # そのため、このメソッドを使用する際には安全性の確保のために以下の引数を指定すること
+    # allowed_targets:　呼び出したいscope名に対応するtargetを過不足なく指定した配列。
+    # default_target: targetに不正な値が渡された際、users_roleが返すスコープ名に対応するtargetを指定する。デフォルトでは:noneを指定しているため何も返さない。
+    def users_role(target, allowed_targets: [], default_target: :none)
+      default_scope = TARGET_TO_SCOPE.fetch(default_target, default_target)
+      scope_name = (ALL_ALLOWED_TARGETS & allowed_targets).include?(target) ? TARGET_TO_SCOPE.fetch(target, target) : default_scope
```

Comment:
```suggestion 
      scope_name = (ALL_ALLOWED_TARGETS & allowed_targets).include?(target) ?
        TARGET_TO_SCOPE.fetch(target, target) : TARGET_TO_SCOPE.fetch(default_target, default_target)
```

でもいいかも（不必要に`TARGET_TO_SCOPE.fetch(default_target, default_target)`を呼び出さないようにする）

Reply:
教えていただいた方法で記述すると、rubocop(Layout/LineLength)に引っかかってしまいました。
`(ALL_ALLOWED_TARGETS & allowed_targets).include?(target)`を切り出す方法も考えましたが、素直に三項演算子をifに書き換える方法を採用しました。
40f6c1e3a6dcce392f6a9324e76874e565926db6

```ruby
    def users_role(target, allowed_targets: [], default_target: :none)
      valid_target = (ALL_ALLOWED_TARGETS & allowed_targets).include?(target)
      scope_name = valid_target ? TARGET_TO_SCOPE.fetch(target, target) : TARGET_TO_SCOPE.fetch(default_target, default_target)
      send(scope_name)
    end
```